### DISCUSSION TITLE
Reason for "Save your work" modal when loading TW from data uri?

### DISCUSSION THREAD

**Discussion by btheado** [TiddlyWiki/TiddlyWiki5#7294] [Q&A] [answered]
I created this data uri:

data:text/html,%0A%3Cscript%3E%0Afetch(%22https%3A%2F%2Ftiddlywiki.com%22)%0A%20%20.then(result%20%3D%3E%20result.text())%0A%20%20.then(html%20%3D%3E%20%7B%0A%20%20%20%20document.open()%3B%0A%20%20%20%20document.write(html)%3B%0A%20%20%20%20document.close()%3B%0A%20%20%7D)%3B%0A%3C%2Fscript%3E%0A

It contains this javascript code:

```js
<script>
fetch("https://tiddlywiki.com")
  .then(result => result.text())
  .then(html => {
    document.open();
    document.write(html);
    document.close();
  });
</script>
```
When I paste the above data uri into the browser bar, it will successfully load tiddlywiki.com. However it pops up a "Save your work" modal dialog box whose instructions I don't really understand. @Jermolene do you recall the reason for this modal?

The modal was introduced in 2012 in this commit: 507460eb4ef9c

Here is the current code which launches this modal:

https://github.com/Jermolene/TiddlyWiki5/blob/2271f6885af8b6301696d3aa33f82472f691ba72/core/modules/startup/rootwidget.js#L90-L96

After closing the modal, I'm able to save changes just fine using the download saver. I didn't need to follow any of the instructions given in the modal. Though the filename of the downloaded file is pretty ugly.

For what use-case does that modal help?

**Comment by Jermolene** [comment, marked as answer]
Hi @btheado good sleuthing! That stuff was to support Safari, which at that time did not support the "download" attribute on the anchor element. Instead, it was possible to create a bookmark from the base64 representation of the wiki.

I think the download attribute has now been supported on Safari for several years, so no doubt it can all be removed now.

**Reply by btheado** [reply]
Thanks for the pointer. If I'm reading this right https://caniuse.com/download, it seems desktop safari has had support since 2017 and mobile safari has had it since 2019. IE doesn't support it.

If it is fine to remove, I can create the PR.

**Reply by Jermolene** [reply]
Thanks @btheado please do go ahead.

Looking through that commit was quite nostalgic, there were a few points of interest:

* Modals are introduced in this commit, meaning that implementing the Mobile Safari saver was actually the trigger that drove the introduction of modals
* The first bullet point of `$:/messages/SaveInstructions` is categorically wrong, I don't think there's ever been a browser for which that has worked. It would take a bit more sifting through the [history](https://github.com/Jermolene/TiddlyWiki5/commits/2271f6885af8b6301696d3aa33f82472f691ba72/core/messages/SaveInstructions.tid) to figure out what was going on
* Back then, the prefix `tw-` was used for everything: including both messages and CSS classes. I do remember it was a bit of a  nightmare to switch them around to the present conventions
